### PR TITLE
ci: add on board pipeline

### PR DESCRIPTION
**Pull request by yihau**
### Problem

Maybe we need a on board pipeline to add people to team.

### Summary of Changes

There are two pipelines. one is for managing team and another is for update account list.

#### Add user to team

Screenshot 2023-02-08 at 11 40 03 PM
Screenshot 2023-02-08 at 11 40 12 PM

We can add a label 'processing' to a qualified issue to trigger this pipeline. The pipeline will:

1. check the label name

2. parse issue's title to get info
 
3. check permission for the person who labeled

4. add users to team and remove users from the team of previous level if needed

5. 
    on success => a) remove 'processing' label b) add 'approved' label with a PAT (for triggering next pipeline)
    on failure => remove 'processing' when failed

#### Update account list

Screenshot 2023-02-11 at 1 45 38 AM

the previous pipeline will add `approved` label to the issue. it will trigger this pipeline.

1. check permisstion (for now I only add Joe cuz I think we will use Joe's PAT)
2. parse issue's body to get `github_username` and `discord_username` and issue's title to know which files need to be modified with the respective level
3. update file and push. there are two situations:
    a) level 1 => add it to the end of the file
    b) level 2 => I suppose it has already on the file so I just change its level by sed. (I won't sort it, I think keep level 3 on the top is enough)
4. close the issue

Before merging this one, we need to setup a `PAT` which's privilege can add people to team (including outside contributors) and add label to issues.

### CODE REVIEW

**Review by joeaba — CHANGES_REQUESTED** [review]

File: .github/workflows/on-board.yml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      - name: check privilege
+        run: |
+          maintainers=(
+            "mvines"
+            "CriesofCarrots"
+            "joeaba"
+            "t-nelson"
+            "sakridge"
+            "yihau"
+            "joncinque"
```
[the remark this answers is not in the document]
Comment (yihau) [reply .github/workflows/on-board.yml:36]: yeah. Thank you for pointing it out. I will move `parse info` step before the permission check.

File: .github/workflows/on-board.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
+name: on-board
+
+on:
+  issues:
+    types:
+      - labeled
+
+jobs:
+  link_to_issue:
+    if: github.event.label.name == 'processing'
```
[the remark this answers is not in the document]
Comment (joeaba) [reply .github/workflows/on-board.yml:10]: right, my bad, makes sense

**Comment by joeaba** [comment]
I just moved the acl from the wiki to #102 so we can can also automate the list, add/remove based on the approved access level

File: .github/workflows/on-board.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yihau) [reply .github/workflows/on-board.yml]: updated the new flow description to the PR body 👻

**Comment by yihau** [comment]
do we want to upgrade the list when changes or build it hourly even daily? Just a little worry about the git conflict

**Comment by yihau** [comment]
add update acl pipeline and some adjustments for the pipeline which adding people to team. More details has updated in the PR's body.

File: .github/workflows/update-acl.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+jobs:
+  check_permission:
+    if: github.event.label.name == 'approved'
+    runs-on: ubuntu-latest
+    steps:
+      - name: check privilege
+        run: |
+          maintainers=(
+            "joeaba"
```
Comment (joeaba) [inline .github/workflows/update-acl.yml:16]: why just me?

Comment (yihau) [reply .github/workflows/update-acl.yml:16]: Ideally the update acl pipeline will be triggered by the on board pipeline. I think we will use your PAT in the on board pipeline so I only write your name here. If we would like to make this step can be triggered by others manually, we can have more names here!

Comment (joeaba) [reply .github/workflows/update-acl.yml:16]: yeah ok that's fine!

**Review by joeaba — APPROVED** [review]

**Pull request merged**